자동차 레이싱 1단계

wishwisdom opened the pull request:
Car 생성
 - 자동차의 상태 및 행동 내용이 함께 들어있는 class

RacingGame
 - 실제 Racing이 일뤄지는 class
 - Car 생성 및 Car의 전진을 결정하는 역할을 함

shinhong-codelinkcorp committed:
Car 생성
 - 자동차의 상태 및 행동 내용이 함께 들어있는 class

RacingGame
 - 실제 Racing이 일뤄지는 class
 - Car 생성 및 Car의 전진을 결정하는 역할을 함

javajigi commented on src/main/java/racing/Main.java:
```diff
@@ -0,0 +1,22 @@
+
+        game = new RacingGame(number, trials);
+
+        game.startRacing();
```
도메인 로직과 View 로직을 구분하기 위해 다음과 같은 구조로 도전해 보면 어떨까요?

```
public static void main(String[] args) {
    int carNo = InputView.getCarNo();
    int tryNo = InputView.getTryNo();

    RacingGame racingGame = new RacingGame(carNo);
    for(int i = 0; i < tryNo; i++) {
        GameResult result = racingGame.move();
        ResultView.print(result);
    }
}
```

GameResult에 이동한 위치 정보가 포함되어 있고요. 이 위치 정보를 활용해 출력하는 구조로 구현해 보면 좋을 것 같아요.

위 코드에서 힌트를 얻어 본인의 코드에 개선할 부분이 있는지 검토해 보고 리팩토링 해보세요.

javajigi commented on src/test/java/racing/RacingGameTest.java:
```diff
@@ -0,0 +1,49 @@
+        game.goStop(car, num);
+
+        assertThat(car.getLocation()).isEqualTo("-");
+
```
불필요하게 공백 라인을 만들지 않는다.
공백 라인을 뛰우는 것도 코드상에 문맥이 달라지는 부분에 의도를 가지고 뛰우면 좋겠다.

javajigi commented on src/main/java/racing/RacingGame.java:
```diff
@@ -0,0 +1,59 @@
+            System.out.println();
+            System.out.println();
+        }
+
```
메소드가 너무 많은 일을 하고 있다. 다음 원칙을 지켜 메소드를 구현한다.

작게 만들어라. - 함수를 만드는 첫 번재 규칙은 '작게'다. 함수를 만드는 두 번째 규칙은 '더 작게'다.
한 가지만 해라. - 함수는 한 가지를 해야 한다. 그 한 가지를 잘 해야 한다. 그 한 가지만 해야 한다.

javajigi commented on src/main/java/racing/RacingGame.java:
```diff
@@ -0,0 +1,59 @@
+        }
+    }
+
+
```
불필요하게 공백 라인을 만들지 않는다.
공백 라인을 뛰우는 것도 코드상에 문맥이 달라지는 부분에 의도를 가지고 뛰우면 좋겠다.

javajigi reviewed (changes requested):
늦었지만 지금이라도 도전한 것을 응원합니다. 최소 2단계까지는 진행할 것을 추천해요. 피드백 남겼어요.

shinhong-codelinkcorp committed:
RacingGame
 - 로직부분만 남겨둠.
 - int trials 삭제
   . Main에서 move 횟수를 결정
 - startRacing() -> readyRacing() 변경
 - checkMinSize, createCars
   . 한 method 안에 한 가지 일만 하게 불리함

 RacingInput
  - main method에서 입력을 불리함
    . 입력은 RacingInput이 담당하도록 변경

 RacingView
  - Car의 정볼르 토대로 경기 상활을 출력함
